tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,418 @@
+import os
+
+import pytest
+import pandas as pd
+import numpy as np
+
+import semeio.workflows.ahm_analysis.ahmanalysis as ahmanalysis
+from semeio._exceptions.exceptions import ValidationError
+
+from unittest.mock import MagicMock
+from itertools import product
+from pathlib import Path
+
+from scipy import stats
+from ecl.grid import EclGridGenerator
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_make_update_log_df(test_data_root):
+    """test function creates a dataframe from update_log file
+    and replace '...' with key_obs"""
+    test_data_dir = os.path.join(test_data_root, "update_log/oneobs")
+    update_log_path = test_data_dir
+    key_obs = "RWI_3_OBS"
+    updatelog_obs = ahmanalysis.make_update_log_df(update_log_path, key_obs)
+    assert "..." not in updatelog_obs["obs_key"]
+    assert updatelog_obs.at[0, "status"] == "Active"
+    assert updatelog_obs.at[1, "status"] == "Inactive"
+    assert (
+        updatelog_obs.columns
+        == [
+            "obs_number",
+            "obs_key",
+            "obs_mean",
+            "obs_std",
+            "status",
+            "sim_mean",
+            "sim_std",
+        ]
+    ).all()
+
+
+@pytest.mark.parametrize(
+    "input_dir, obs_key, expected_result",
+    [
+        ("update_log/oneobs", "RWI_3_OBS", "2 active/3"),
+        ("update_log/allobs", "ALL_OBS", "41 active/47"),
+        ("update_log/noactive", "RWI_2_OBS", "0 active/3"),
+    ],
+)
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_list_active_observations(test_data_root, input_dir, expected_result, obs_key):
+    """test function creates a dataframe reporting active observations during update"""
+    test_data_dir = os.path.join(test_data_root, input_dir)
+    df_update_log = ahmanalysis.make_update_log_df(test_data_dir, obs_key)
```

Comment:
Reading this test, I wonder.. The `make_update_log_df` would yield:

```
35         36    RWI_3_OBS    285.00     3.00    Active   273.958   10.529
36         37      ALL_OBS    258.00     3.00    Active   264.119    9.031
37         38      ALL_OBS    271.00     3.00    Active   269.818    5.951
38         39    RWI_2_OBS    291.00     3.00    Active   272.672   14.870
39         40      ALL_OBS    233.00     3.00    Active   265.408   14.494
40         41      ALL_OBS    297.00     3.00    Active   277.500   13.479
41         42    ROP_5_OBS    285.00     3.00    Active   274.673    6.869
42         43      ALL_OBS    279.00     3.00    Active   271.760    5.229
43         44      ALL_OBS    288.00     3.00    Active   270.670    5.802
44         45    ROP_4_OBS    292.00     3.00    Active   279.577    8.164
45         46      ALL_OBS    268.00     3.00    Active   273.585    6.250
46         47      ALL_OBS    280.00     3.00    Active   273.649    6.049
```

For the input:

 ```
 36  : RWI_3_OBS                                285.000 +/-            3.000  Active   |           273.958 +/-          10.529
  37  :   ...                                    258.000 +/-            3.000  Active   |           264.119 +/-           9.031
  38  :   ...                                    271.000 +/-            3.000  Active   |           269.818 +/-           5.951
  39  : RWI_2_OBS                                291.000 +/-            3.000  Active   |           272.672 +/-          14.870
  40  :   ...                                    233.000 +/-            3.000  Active   |           265.408 +/-          14.494
  41  :   ...                                    297.000 +/-            3.000  Active   |           277.500 +/-          13.479
  42  : ROP_5_OBS                                285.000 +/-            3.000  Active   |           274.673 +/-           6.869
  43  :   ...                                    279.000 +/-            3.000  Active   |           271.760 +/-           5.229
  44  :   ...                                    288.000 +/-            3.000  Active   |           270.670 +/-           5.802
  45  : ROP_4_OBS                                292.000 +/-            3.000  Active   |           279.577 +/-           8.164
  46  :   ...                                    268.000 +/-            3.000  Active   |           273.585 +/-           6.250
  47  :   ...                                    280.000 +/-            3.000  Active   |           273.649 +/-           6.049
```

Is that the intention?

Reply:
as explained before initially yes when there was the bug in the updatelog file not reporting missing data. But I agree the output now should use ffill and the ... are to be replaced by the correct obs name.